python/fbprophet/plot.py

```diff
@@ -136,9 +136,10 @@ def plot_components(
                 m=m, fcst=fcst, name='trend', ax=ax, uncertainty=uncertainty,
                 plot_cap=plot_cap,
             )
-        elif plot_name == 'weekly':
+        elif plot_name == 'weekly' or (plot_name in m.seasonalities and
+                                       m.seasonalities[plot_name]['period'] == 7):
```

Comment:
Instead of modifying plot_weekly, we could remove the added condition here and always pass it to  plot_forecast_component instead. It's however a bit confusing in this case since weekdays are not visible in plot_forecast_component.

Reply:
Yeah this is nice. We should probably do a little refactoring here and just get rid of the `plot_name == 'weekly'` check altogether and just have it use `plot_weekly` if period==7 as is done here, and the same for yearly `if period ==365.25`.  Then we can delete L106-L111, and replace L115 with `if name in fcst`

Reply:
Seems like L106-L111 was added to keep order same, so not sure if I should chnage that. Did some light refactoring here, kept plot_name == 'weekly'  and 'yearly to avoid changing behavior if someone has modified the period, but that's probably quite rare.